File: MatskocaAdmin/MatskocaAdmin/API/Models/WorkModels.swift
import Foundation

struct MetricUi: Identifiable {
    let id = UUID()
    let label: String
    let value: String
}

struct ListCardUi: Identifiable {
    let id = UUID()
    let title: String
    var subtitle: String = ""
    var meta: String = ""
    var badge: String?
    var badgeColor: BadgeColor = .neutral
    var clientId: Int?
    var ticketId: Int?
    var trainingId: String?
    var trainingDate: String?
    var refType: String?
    var feedId: Int?

    var isClickable: Bool {
        clientId != nil || ticketId != nil || trainingId != nil
            || refType == "client" || refType == "ticket"
    }
}

struct HomeSectionUi: Identifiable {
    let id = UUID()
    let title: String
    let items: [ListCardUi]
    var emptyMessage: String?
}

struct ProfileSectionUi: Identifiable {
    let id = UUID()
    let key: String
    let title: String
    var hint: String = ""
}

enum BadgeColor {
    case neutral, success, warning, error, primary
}

struct ActionUi: Identifiable {
    let id: String
    let label: String
}

struct DayChipUi: Identifiable {
    let id = UUID()
    let date: String
    let label: String
    let count: Int
    let selected: Bool
}

struct HomeTabUi {
    var greeting: String = ""
    var roleTitle: String = ""
    var metrics: [MetricUi] = []
    var showAdminButton: Bool = false
    var showEntryQr: Bool = false
    var entryQrStaffUserId: Int = 0
    var entryQrActive: Bool = false
    var entryQrBlockedMessage: String?
    var entryQrFormat: String = "ascii"
    var needNotificationsPermission: Bool = false
    var sections: [HomeSectionUi] = []
    var sectionTitle: String?
    var items: [ListCardUi] = []
    var actions: [ActionUi] = []
    var loading: Bool = false
    var emptyMessage: String?
}

struct ScheduleDayUi: Identifiable {
    let id = UUID()
    let date: String
    let weekdayLabel: String
    let dayNumber: String
    let sessionCount: Int
    let selected: Bool
    let isToday: Bool
}

struct ScheduleSessionUi: Identifiable {
    let id = UUID()
    var trainingId: String?
    var date: String = ""
    let title: String
    let type: String
    let typeLabel: String
    let startTime: String
    let endTime: String
    let durationMinutes: Int
    let trainer: String
    let room: String
    var bookedCount: Int?
    var maxParticipants: Int?
    var clientNames: [String] = []
    var bookings: [ScheduleBookingUi] = []
}

struct ScheduleBookingUi: Identifiable {
    let id: String
    let clientName: String
    var clientId: Int?
}

struct ScheduleTabUi {
    var days: [ScheduleDayUi] = []
    var sessions: [ScheduleSessionUi] = []
    var monthLabel: String = ""
    var selectedTypeFilter: String?
    var denied: Bool = false
    var deniedMessage: String = ""
    var loading: Bool = false
}

struct AssignClientDialogUi {
    var trainingId: String
    var sessionTitle: String
    var query: String = ""
    var clients: [ListCardUi] = []
    var booked: [ListCardUi] = []
    var loading: Bool = false
    var errorMessage: String?
}

struct ClientsTabUi {
    var query: String = ""
    var summary: String = ""
    var items: [ListCardUi] = []
    var onlyActiveBooking: Bool = false
    var denied: Bool = false
    var deniedMessage: String = ""
    var loading: Bool = false
}

struct SupportTabUi {
    var newCount: Int = 0
    var unreadCount: Int = 0
    var filters: [DayChipUi] = []
    var notifications: [ListCardUi] = []
    var tickets: [ListCardUi] = []
    var ticketActions: [Int: [ActionUi]] = [:]
    var actions: [ActionUi] = []
    var denied: Bool = false
    var deniedMessage: String = ""
    var loading: Bool = false
}

struct ProfileTabUi {
    var name: String = ""
    var email: String = ""
    var roleTitle: String = ""
    var phone: String = ""
    var specialization: String = ""
    var description: String = ""
    var photoUrl: String?
    var sections: [ProfileSectionUi] = []
    var adminAvailable: Bool = false
    var showAdminButton: Bool = false
    var showTrainerProfileEdit: Bool = false
    var showClubEntryQr: Bool = false
    var showRentalManage: Bool = false
    var showFeedback: Bool = true
    var rentalPaidUntilLabel: String?
    var paidRentalClubs: [RentalClubOption] = []
    var activeClubId: Int?
    var offerUrl: String = "https://dobrozal.ru/doc/offer"
    var privacyUrl: String = "https://dobrozal.ru/doc/privacy"
    var docsUrl: String = "https://dobrozal.ru/doc"
    var sectionTitle: String?
    var items: [ListCardUi] = []
    var loading: Bool = false
}

enum WorkTab: Int {
    case home = 1
    case schedule = 2
    case clients = 3
    case profile = 4
    case support = 5

    var title: String {
        switch self {
        case .home: return "Главная"
        case .schedule: return "Расписание"
        case .clients: return "Клиенты"
        case .profile: return "Профиль"
        case .support: return "Обращения"
        }
    }
}

struct WorkUiState {
    var selectedTab: WorkTab = .home
    var screenTitle: String = "Главная"
    var statusMessage: String?
    var errorMessage: String?
    var showScheduleNav: Bool = false
    var showClientsNav: Bool = false
    var showSupportNav: Bool = false
    var home: HomeTabUi = HomeTabUi()
    var schedule: ScheduleTabUi = ScheduleTabUi()
    var clients: ClientsTabUi = ClientsTabUi()
    var support: SupportTabUi = SupportTabUi()
    var profile: ProfileTabUi = ProfileTabUi()
    var assignDialog: AssignClientDialogUi?
}

struct RoleOptionUi: Identifiable, Equatable {
    var id: String { role }
    let label: String
    let role: String
}

struct LoginUiState {
    var email: String = ""
    var name: String = ""
    var password: String = ""
    var selectedRole: RoleOptionUi?
    var roles: [RoleOptionUi] = []
    var configSummary: String = ""
    var statusMessage: String?
    var errorMessage: String?
    var isLoading: Bool = false
}

struct AdminHubUi {
    var canWrite: Bool = false
    var metrics: [MetricUi] = []
    var sections: [AdminSectionRowUi] = []
    var loading: Bool = true
    var error: String?
}

struct AdminSectionRowUi: Identifiable {
    let id = UUID()
    let key: String
    let title: String
    let hint: String
}

struct AdminSectionUi {
    var title: String = ""
    var metrics: [MetricUi] = []
    var items: [ListCardUi] = []
    var shortcuts: [ActionUi] = []
    var summary: String = ""
    var loading: Bool = true
    var error: String?
}

enum ClientBookingTab {
    case active, completed
}

struct ClientDetailUi {
    var title: String = "Клиент"
    var name: String = ""
    var email: String = ""
    var phone: String = ""
    var bonusPoints: Int = 0
    var isBlocked: Bool = false
    var subscriptionTitle: String = ""
    var subscriptionMeta: String = ""
    var activeBookings: [ListCardUi] = []
    var completedBookings: [ListCardUi] = []
    var bookingTab: ClientBookingTab = .active
    var tickets: [ListCardUi] = []
    var loading: Bool = true
    var error: String?
    var showCallButton: Bool = false
}

File: staffapp/src/main/java/com/example/staffapp/ui/work/WorkModels.kt
package com.example.staffapp.ui.work

import com.example.staffapp.RentalClubOption

data class MetricUi(val label: String, val value: String)

data class ListCardUi(
    val title: String,
    val subtitle: String = "",
    val meta: String = "",
    val badge: String? = null,
    val badgeColor: BadgeColor = BadgeColor.NEUTRAL,
    val clientId: Int? = null,
    val ticketId: Int? = null,
    val refType: String? = null,
    val feedId: Int? = null,
    val trainingId: String? = null,
    val trainingDate: String? = null,
) {
    val isClickable: Boolean
        get() = clientId != null || ticketId != null || trainingId != null ||
            refType == "client" || refType == "ticket"
}

data class ProfileSectionUi(
    val key: String,
    val title: String,
    val hint: String = "",
)

enum class BadgeColor { NEUTRAL, SUCCESS, WARNING, ERROR, PRIMARY }

data class ActionUi(val id: String, val label: String)

data class DayChipUi(val date: String, val label: String, val count: Int, val selected: Boolean)

data class HomeSectionUi(
    val title: String,
    val items: List<ListCardUi> = emptyList(),
    val emptyMessage: String? = null,
)

data class HomeTabUi(
    val greeting: String = "",
    val roleTitle: String = "",
    val metrics: List<MetricUi> = emptyList(),
    val showAdminButton: Boolean = false,
    /** QR прохода на главной (тренер с активной арендой). */
    val showEntryQr: Boolean = false,
    val entryQrStaffUserId: Int = 0,
    val entryQrActive: Boolean = false,
    val entryQrBlockedMessage: String? = null,
    val sections: List<HomeSectionUi> = emptyList(),
    val actions: List<ActionUi> = emptyList(),
    val loading: Boolean = false,
    val needNotificationsPermission: Boolean = false,
)

data class ScheduleDayUi(
    val date: String,
    val weekdayLabel: String,
    val dayNumber: String,
    val sessionCount: Int,
    val selected: Boolean,
    val isToday: Boolean = false,
)

data class ScheduleSessionUi(
    val trainingId: String? = null,
    val date: String = "",
    val title: String,
    val type: String,
    val typeLabel: String,
    val startTime: String,
    val endTime: String,
    val durationMinutes: Int,
    val trainer: String,
    val room: String,
    val bookedCount: Int? = null,
    val maxParticipants: Int? = null,
    val clientNames: List<String> = emptyList(),
    val bookings: List<ScheduleBookingUi> = emptyList(),
)

data class ScheduleBookingUi(
    val id: String,
    val clientName: String,
    val clientId: String? = null,
)

data class ScheduleTabUi(
    val days: List<ScheduleDayUi> = emptyList(),
    val sessions: List<ScheduleSessionUi> = emptyList(),
    val monthLabel: String = "",
    val selectedTypeFilter: String? = null,
    val denied: Boolean = false,
    val deniedMessage: String = "",
    val loading: Boolean = false,
)

data class ClientsTabUi(
    val query: String = "",
    val summary: String = "",
    val items: List<ListCardUi> = emptyList(),
    /** Фильтр «только с активной записью» (пункт 31 репорта). */
    val onlyActiveBooking: Boolean = false,
    val denied: Boolean = false,
    val deniedMessage: String = "",
    val loading: Boolean = false,
)

data class SupportTabUi(
    val newCount: Int = 0,
    val unreadCount: Int = 0,
    val filters: List<DayChipUi> = emptyList(),
    val notifications: List<ListCardUi> = emptyList(),
    val tickets: List<ListCardUi> = emptyList(),
    val ticketActions: Map<Int, List<ActionUi>> = emptyMap(),
    val actions: List<ActionUi> = emptyList(),
    val denied: Boolean = false,
    val deniedMessage: String = "",
    val loading: Boolean = false,
)

data class ProfileTabUi(
    val name: String = "",
    val email: String = "",
    val roleTitle: String = "",
    val phone: String = "",
    val specialization: String = "",
    val description: String = "",
    val photoUrl: String? = null,
    /** Действующая аренда клуба (п.25). */
    val rentalPaidUntilLabel: String? = null,
    /** Оплаченные залы для смены адреса. */
    val paidRentalClubs: List<RentalClubOption> = emptyList(),
    val activeClubId: Int? = null,
    val showClubEntryQr: Boolean = false,
    val showRentalManage: Boolean = false,
    val showFeedback: Boolean = false,
    /** Ссылки на документы (п.26). */
    val offerUrl: String = "https://dobrozal.ru/doc/offer",
    val privacyUrl: String = "https://dobrozal.ru/doc/privacy",
    val docsUrl: String = "https://dobrozal.ru/doc",
    val sections: List<ProfileSectionUi> = emptyList(),
    val adminAvailable: Boolean = false,
    val showAdminButton: Boolean = false,
    val showTrainerProfileEdit: Boolean = false,
    val sectionTitle: String? = null,
    val items: List<ListCardUi> = emptyList(),
    val loading: Boolean = false,
)

data class WorkUiState(
    val selectedTab: Int = TAB_HOME,
    val screenTitle: String = "Главная",
    val statusMessage: String? = null,
    val errorMessage: String? = null,
    val showScheduleNav: Boolean = false,
    val showClientsNav: Boolean = false,
    val showSupportNav: Boolean = false,
    val home: HomeTabUi = HomeTabUi(),
    val schedule: ScheduleTabUi = ScheduleTabUi(),
    val clients: ClientsTabUi = ClientsTabUi(),
    val support: SupportTabUi = SupportTabUi(),
    val profile: ProfileTabUi = ProfileTabUi(),
    val assignDialog: AssignClientDialogUi? = null,
    val createSessionDialog: CreateSessionDialogUi? = null,
) {
    companion object {
        const val TAB_HOME = 1
        const val TAB_SCHEDULE = 2
        const val TAB_CLIENTS = 3
        const val TAB_PROFILE = 4
        const val TAB_SUPPORT = 5
    }
}
